### PR TITLE
Cater for sql files in nested folders

### PR DESCRIPTION
**Pull request by yongchongye**
Did a quick and dirty fix to switch to the new [`os.DirFS`](https://pkg.go.dev/os#DirFS) instead of `http.Dir`
Also using `fs.WalkDir` to collect the `.sql` files

Further enhancements
- the up and down sql files could be put into different slices during `fs.WalkDir`

### CODE REVIEW

File: lib.go [hunk 1]
```diff
@@ -95,13 +95,22 @@ func New(dir http.FileSystem, driverName string, databaseURL string) (*Config, e
...
+	var migrationFiles []string
+	err = fs.WalkDir(dir, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		fp := path
+		if !strings.HasSuffix(path, ".sql") &&
```
Comment (yongchongye) [inline lib.go:107]: this is the main check catering for nested folders

File: lib.go [hunk 2]
```diff
@@ -192,14 +200,13 @@ func (c *Config) MigrateUp(ctx context.Context, txOpts *sql.TxOptions, logFilena
 
 	migrationFiles := c.migrationFiles
 	sort.SliceStable(migrationFiles, func(i int, j int) bool {
```
Comment (yongchongye) [inline lib.go:202]: might not need this anymore, as `fs.WalkDir` is in lexical order

File: lib.go [hunk 3]
```diff
@@ -243,13 +250,12 @@ func (c *Config) MigrateDown(ctx context.Context, txOpts *sql.TxOptions, logFile
...
```
Comment (yongchongye) [inline lib.go:252]: might not need this anymore, as `fs.WalkDir` is in lexical order

File: lib.go [hunk 4]
```diff
@@ -313,7 +319,7 @@ type Adapter struct {
 
 var adapters = map[string]Adapter{
 	"postgres": {
-		CreateVersionsTable:    `CREATE TABLE dbmigrate_versions (version char(14) NOT NULL PRIMARY KEY)`,
+		CreateVersionsTable:    `CREATE TABLE IF NOT EXISTS dbmigrate_versions (version char(14) NOT NULL PRIMARY KEY)`,
```
Comment (choonkeat) [inline lib.go:322]: forgot if there was a reason to skip it? maybe it was just an omission

**Review by choonkeat — APPROVED** [review]
could only verify this after https://github.com/choonkeat/dbmigrate/pull/12

LGTM

**Pull request merged**